x-pack/plugins/ml/server/models/results_service/get_partition_fields_values.ts

```diff
@@ -77,6 +77,17 @@ function getFieldAgg(
                   },
                 ]
               : []),
+            ...(fieldConfig?.filterBy
```

Comment:
I do not think we need an additional param. There is already the `searchTerm` argument. Just make sure it is populated correctly and filters are applied.